compose-builder/tui-generator.sh

```diff
@@ -0,0 +1,299 @@
+    do
+        arglist+=("$index")
+        arglist+=("${deviceServiceDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_DEVSERVICES=$($WHIPTAIL --title "Device Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --checklist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Message Bus Options Display Function                          #
+####################################################################
+function msgBusOption() {
+    message="Available Message Buses to replace the default one.\n
+            Press <SPACEBAR> to Select \n
+            Press <ENTER> to Skip
+            "
+
+    # Generate a Specific form of Array required by whiptail checklist
+    # FORMAT: "<INDEX> <DESCRIPTION> <OFF>"
+    arglist=()
+    for index in "${messageBus[@]}";
+    do
+        arglist+=("$index")
+        arglist+=("${msgBusDesc[$index]}")
+        arglist+=("OFF") # Nothing is selected by-default
+    done
+    SELECTED_BUS=$($WHIPTAIL --title "App Services" \
+                --ok-button Next \
+                --nocancel \
+                --notags --separate-output \
+                --radiolist "$message" $LINES $COLUMNS $(( $LINES - 12 )) \
+                -- "${arglist[@]})" \
+                3>&1 1>&2 2>&3)
+}
+
+####################################################################
+#    Generate / Generate & Run Option Display Function             #
+####################################################################
+function finalStep() {
+    message="What would you like to do?"
+    $WHIPTAIL --title "Non-Secure Configuration" \
+        --yes-button "Generate File" \
+        --no-button "Generate File and Run" \
+         --yesno "$message" $LINES $COLUMNS
+}
+
+## Step - 1: Start by Asking about whether Images should be pulled for ARM64?
+displayArm64
+if [[ $? == 0 ]]; then
+    SELECTED_OTHERS+=("arm64 ")
+fi
+
+## Step - 2: Ask whether a non-secure Configuration is needed
+displayNoSecty
+if [[ $? == 0 ]]; then
+    SELECTED_OTHERS+=("no-secty ")
+fi  
+
+## Step - 3: Add Device Services Option
+devServiceOption
+
+## Step - 4: App Services Option
+appServiceOption
+
+## Step - 5: Configure Message Bus
+msgBusOption
+
+## Step -6: Ask for Additional Options
+additionalServiceOption
+
+# echo "additional services selected: ${SELECTED_OTHERS}" ## DEBUG
+# echo "device services selected: ${SELECTED_DEVSERVICES}" ## DEBUG
+# echo "app services selected: ${SELECTED_APPSERVICES}" ## DEBUG
+# echo "selected message bus: ${SELECTED_BUS}" ## DEBUG
+
+finalStep
+
+if [[ $? == 0 ]]; then
+    echo "Generating Compose file...\n"
+    $MAKE gen $SELECTED_OTHERS $SELECTED_DEVSERVICES $SELECTED_APPSERVICES $SELECTED_BUS
```

Comment:
Since these are arrays, you should reference them as such:

```suggestion
    $MAKE gen "${SELECTED_OTHERS[@]}" "${SELECTED_DEVSERVICES[@]}" "${SELECTED_APPSERVICES[@]}" "${SELECTED_BUS[@]}"
```

Reply:
Now that I look closer, they are not actually arrays, they were just defined at the top as such, so doing it this way will fail. They should either be redefined as strings or read into an array from whiptail's response.